core/client/fs/src/main/java/alluxio/client/block/stream/BlockWorkerClient.java

```diff
@@ -48,20 +48,15 @@ class Factory {
     /**
      * Creates a new block worker client.
      *
-     * @param userState the user subject
+     * @param context the client context
      * @param address the address of the worker
      * @param alluxioConf Alluxio configuration
      * @return a new {@link BlockWorkerClient}
      */
-    public static BlockWorkerClient create(UserState userState, GrpcServerAddress address,
-        AlluxioConfiguration alluxioConf)
-        throws IOException {
-      try {
-        return new DefaultBlockWorkerClient(userState, address, alluxioConf);
-      } catch (Exception e) {
-        throw new IOException(
-            String.format("Failed to connect to block worker (%s)", address), e);
-      }
+    public static BlockWorkerClient create(ClientContext context,
+                                           GrpcServerAddress address,
+                                           AlluxioConfiguration alluxioConf) {
+      return new DefaultBlockWorkerClient(context, address, alluxioConf);
```

Comment:
Since we get rid of the connecting logic in the constructor of `DefaultBlockWorker`, it's impossible to have any exception now.